### ISSUE TITLE
Tracking Issue for `const_maybe_uninit_zeroed`

### ISSUE THREAD

**Issue by woppopo** [rust-lang/rust#91850]
Feature gate: `#![feature(const_maybe_uninit_zeroed)]`

This is a tracking issue for implementing const `MaybeUninit::zeroed`.

### Public API

```rust
// core::mem
impl<T> MaybeUninit<T> {
    pub const fn zeroed() -> MaybeUninit<T>;
}
```

### Steps / History

- [ ] Implementation: yet
- [ ] Final comment period (FCP)
- [ ] Stabilization PR

### Unresolved Questions

- None yet.

**Comment by joshlf** [comment]
Is this waiting for anything before being stabilized?

**Comment by woppopo** [comment]
Yes. This is waiting for `const_ptr_write`(#86302) and `const_maybe_uninit_as_mut_ptr`(#75251).

**Comment by dtolnay** [comment]
Could someone with a use case for this please help out with a compelling example for the docs in https://github.com/rust-lang/rust/pull/116218#pullrequestreview-1650086702? Thanks!
